cryostatio/cryostat-helm
# feat(deployment): split out -db and -storage deployments

Related to #115

This PR targets a new upstream feature branch `separate-db-storage`, which will close out the issue above when merged.

This particular PR just lays the foundation, splitting up the Deployment into three, with each new one gaining its own Service, PVC, etc. as needed, as well as adjusting tests.

To test on OpenShift:

```
helm install cryostat --set authentication.openshift.enabled=true --set core.route.enabled=true --set core.image.tag=3.0.0 --set pvc.enabled=true ./charts/cryostat/
```

(the `core.image.tag` is set to work around https://github.com/cryostatio/cryostat/pull/606 for now)

This should result in a working and accessible Cryostat instance which looks the same as before from an end user perspective. All other `values.yaml` settings should still work the same as before as well.

This does not yet add auth proxies, TLS configurations, network policies, separate service accounts, etc. to enhance the security of the separate deployments. That will be done in a follow-up PR(s).
